### ISSUE TITLE
got CRC errors after using MMKV 1.3.1 with MMKV_DISABLE_CRYPT

### ISSUE THREAD

**Issue by samlu** [Tencent/MMKV#1142]
Got CRC checking errors after upgrading to MMKV 1.3.1 with `MMKV_DISABLE_CRYPT` and `MMKV_DISABLE_FLUTTER` defined.

This problem can be reproduced by the attached sample app.
[test_mmkv_multi_process_v3.zip](https://github.com/Tencent/MMKV/files/12329123/test_mmkv_multi_process_v3.zip)

Please read test_mmkv_multi_process/readme.txt for how to reproduce it and get more detailed information.

How to reproduce the problem
===============================
1. compile the app (I.e. compile the app with MMKV 1.2.15)
2. run it on Android 6 emulator
3. tap the bottom "Press me" button
4. exit from the app
5. edit app/build.gradle
6. comment line 39 and uncomment line 40
7. compile the app (I.e. compile the app with mmkv-StaticCpp-release-1.3.1.noaes.aar)
8. run it on Android 6 emulator and you will see the following CRC checking error

LevelError MMKV.cpp:397 checkFileCRCValid, check crc [fg_single_process] fail, crc32:395274559, m_crcDigest:3003654148
LevelError MMKV.cpp:397 checkFileCRCValid, check crc [fg_single_process] fail, crc32:135604869, m_crcDigest:3003654148
LevelError MMKV_IO.cpp:240 operator(), check [fg_single_process] error: lastActualSize 149, lastActualCRC 135604869
onMMKVCRCCheckFail(fg_single_process)

PS: **mmkv-StaticCpp-release-1.3.1.noaes.aar** is MMKV 1.3.1 but exclude AES and flutter (I.e. define MMKV_DISABLE_CRYPT and MMKV_DISABLE_FLUTTER)

has CRC errors on
===============================
. Android 6 emulator
. Nexus with Android 6.0.1
. Samsung Note 4 with Android 6.0.1

no CRC errors on
===============================
. Nexus 6 with Android 7.1.1
. Android 8 emulator
. Samsung S8 with Android 9
. Pixel with Android 10
. Pixel 3 XL with Android 12

**Comment by lingol** [comment]
I have fail to reproduce it on my device.  Looks like it's a problem only happens on Android 6. 
Could you diagnose it and make a PR for that?

**Comment by lingol** [comment]
Update: I have successfully reproduce it on Android 6 simulator. The problem is that with my local build (
1. define MMKV_DISABLE_CRYPT and MMKV_DISABLE_FLUTTERand upgrade the version to v1.3.2 to make sure no conflict with v1.3.1, and publish to local maven), the error is gone.

**Comment by samlu** [comment]
That's weird. I did the same thing as you did with the dev_with_no_aes_no_flutter branch.

Would you upload the mmkv aar of dev_with_no_aes_no_flutter to this ticket?

**Comment by lingol** [comment]
I'm not quite sure you did the same thing. I have noticed that your aar will print no logs on the default MMKV setting (no handler). Maybe you have done something more?
Anyway, here is the whole package of my build.
[1.3.2.zip](https://github.com/Tencent/MMKV/files/12331382/1.3.2.zip)

**Comment by samlu** [comment]
I should have found the cause!

In addition to define `MMKV_DISABLE_CRYPT `and `MMKV_DISABLE_FLUTTER`, I also change the cmake version to 3.22.1
Rolling back the cmake version to 3.18.1 eliminates the CRC errors on Android 6.

The reason why I use cmake 3.22.1 is it shows the following errors while compiling mmkv with cmake 3.18.1.
```
C/C++: CMake Warning at E:/Android/Sdk/ndk/25.0.8775105/build/cmake/android-legacy.toolchain.cmake:415 (message):
C/C++:   An old version of CMake is being used that cannot automatically detect
C/C++:   compiler attributes.  Compiler identification is being bypassed.  Some
C/C++:   values may be wrong or missing.  Update to CMake 3.19 or newer to use
C/C++:   CMake's built-in compiler identification.
```

FYI.

**Comment by lingol** [comment]
That's strange. I will check on it later.

**Comment by lingol** [comment]
My local cmake version is 3.26.3, FYI.

**Comment by samlu** [comment]
Which cmake version you are using when building the mmkv to Maven?

**Comment by lingol** [comment]
> My local cmake version is 3.26.3, FYI.

This version.

**Comment by lingol** [comment]
Can you confirm that v3.26.3 works? If so, we can move on from this issue and stick to v3.26.3 and above.

**Comment by samlu** [comment]
No, cmake v3.26.3 does not work.

I do some more tests on the Android 6 emulator with different cmake versions, and here is the result:

No CRC error
```
3.18.1
3.19.8
3.20.6
3.21.7
```

Has CRC errors
```
3.22.1
3.26.3
3.26.5
3.27.2
```

It looks like it's ok for you to move the cmake version to 3.21.7, but not 3.22 or higher.

**Comment by lingol** [comment]
I think there's a better solution. How about an official package named `com.tencent:mmkv-tiny` starting from v1.3.2?

**Comment by lingol** [comment]
Frankly, I don't think it's CMake's problem, for v3.26.3 works on my machine but (it and higher) not on yours.

**Comment by samlu** [comment]
I like the idea of `com.tencent:mmkv-tiny`!
 
Would you send me the mmkv-StaticCpp-release.aar (build on your env with cmake 3.26.3) before you release mmkv v1.3.2 to the public? I can help to verify.

**Comment by lingol** [comment]
Sure. I will update it here when that time comes.

**Comment by samlu** [comment]
I also found out that the aar size is 651 KB for those versions that didn't work. FYI.

<img width="295" alt="mmkv_crc" src="https://github.com/Tencent/MMKV/assets/1866938/21428fd6-4398-44e8-9a54-3ba1deebcea5">

**Comment by lingol** [comment]
Released with [v1.3.2](https://github.com/Tencent/MMKV/releases/tag/v1.3.2). 
I'm sorry that the proposed `com.tencent:mmkv-tiny' is not there. If you have the time to make it happen, PRs are always welcome.
